flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/internal/TransactionAbortStrategyImpl.java

```diff
@@ -0,0 +1,215 @@
+                    // from reading)
+                    // This method will only cease to work if transaction log timeout = topic
+                    // retention
+                    // and a user didn't restart the application for that period of time. Then the
+                    // first
+                    // transactions would vanish from the topic while later transactions are still
+                    // lingering until they are cleaned up by Kafka. Then the user has to wait until
+                    // the
+                    // other transactions are timed out (which shouldn't take too long).
+                    break;
+                }
+            }
+            return numTransactionAborted;
+        }
+    },
+    LISTING {
+        @Override
+        public void abortTransactions(Context context) {
+            Collection<String> openTransactionsForTopics = context.getOpenTransactionsForTopics();
+
+            if (openTransactionsForTopics.isEmpty()) {
+                return;
+            }
+
+            List<String> openTransactionsForSubtask =
+                    openTransactionsForTopics.stream()
+                            // ignore transactions from other applications
+                            .filter(name -> hasKnownPrefix(name, context))
+                            //  look only at transactions owned by this subtask
+                            .filter(context::ownsTransactionalId)
+                            .collect(Collectors.toList());
+
+            LOG.warn(
+                    "Found {} open transactions for subtask {}: {}",
+                    openTransactionsForSubtask.size(),
+                    context.getCurrentSubtaskId(),
+                    openTransactionsForSubtask);
+            // look only at transactions coming from this application
+            // remove transactions that are owned by the committer
+            TransactionAborter transactionAborter = context.getTransactionAborter();
+            for (String name : openTransactionsForSubtask) {
+                if (context.getPrecommittedTransactionalIds().contains(name)) {
+                    LOG.debug(
+                            "Skipping transaction {} because it's in the list of transactions to be committed",
+                            name);
+                    continue;
+                }
+                transactionAborter.abortTransaction(name);
+            }
+        }
+
+        private boolean hasKnownPrefix(String name, Context context) {
+            return context.getPrefixesToAbort().stream().anyMatch(name::startsWith);
+        }
+    };
+
+    private static final Logger LOG = LoggerFactory.getLogger(TransactionAbortStrategyImpl.class);
+
+    /** Aborts all transactions that have been created by this subtask in a previous run. */
+    public abstract void abortTransactions(Context context);
+
+    /** Injects the actual abortion of the transactional id generated by one of the strategies. */
```

Comment:
Please could we use the word `abort` `aborted` `aborting`, which seems to be the words used in reference to transactions. I think we should avoid `abortion` in any text or variable/method name, as it could be triggering.